src/s1reader/s1_reader.py

```diff
@@ -526,6 +528,80 @@ def get_track_burst_num(track_burst_num_file: str = esa_track_burst_id_file):
+
+    Parameters
+    ----------
+    orbit_state_vector_list: ET
+        XML elements that points to the list of orbit information.
+        each elements in the parameter contains the information below:
+        TAI, UTC, UT1, Absolute_Orbit, X, Y, Z, VX, VY, VZ,Quality
+
+    sensing_time: datetime.datetime
+        Sensing time of the data
+
+    Returns
+    -------
+    _ : datetime.datetime
+        Ascending node crossing time calculated from orbit
+    '''
+
+    # Crop the orbit information before 2 * (orbit period) of sensing time
+    search_length = datetime.timedelta(seconds=2 * T_ORBIT)
+    orbit_until_sensing_time = get_burst_orbit(sensing_time  - search_length,
+                                               sensing_time,
+                                               orbit_state_vector_list)
+
+    # Convert ISCE3 oribt object's reference datetime into python object
+    datetime_orbit_ref = datetime.datetime(
+        orbit_until_sensing_time.reference_epoch.year,
+        orbit_until_sensing_time.reference_epoch.month,
+        orbit_until_sensing_time.reference_epoch.day)
+    datetime_orbit_ref += datetime.timedelta(
+        seconds=orbit_until_sensing_time.reference_epoch.seconds_of_day())
+
+    # detect the event of ascending node crossing from the cropped orbit info.
+    # Algorithm inspirates by Scott Staniewicz's PR in the link below:
+    # https://github.com/opera-adt/s1-reader/pull/120/
+    datetime_ascending_node_crossing_list = []
+    orbit_time_vec = np.array(orbit_until_sensing_time.time)
+    orbit_z_vec = orbit_until_sensing_time.position[:,2]
+
+    # Iterate through the z coordinate in orbit object to
+    # detect the ascending node crossing
+    z_prev = None
+    for index_z, z in enumerate(orbit_z_vec):
+        if z_prev is not None and (z_prev < 0 <= z):
+            # Extract z coords. and time for interpolation
+            index_from = max(index_z - 3, 0)
+            index_to = min(index_z + 3, len(orbit_z_vec))
+            z_around_crossing = orbit_z_vec[index_from : index_to]
+            time_around_crossing = orbit_time_vec[index_from : index_to]
+
+            # Set up spline interpolator and interpolate the time when z is equal to 0.0
+            interpolator_time = InterpolatedUnivariateSpline(z_around_crossing,
+                                                             time_around_crossing,
+                                                             k=1)
+            t_interp = interpolator_time(0.0)
+
+            # Convert the interpolated time into datetime
+            # Add the ascending node crossing time if it's before the sensing time
+            datetime_ascending_crossing = (datetime_orbit_ref
+                                           + datetime.timedelta(seconds=float(t_interp)))
+            if datetime_ascending_crossing < sensing_time:
+                datetime_ascending_node_crossing_list.append(datetime_ascending_crossing)
+        z_prev = z
```

Comment:
```suggestion
        if z_prev is None or not (z_prev < 0 <= z):
            z_prev = z
            continue
        # Extract z coords. and time for interpolation
        index_from = max(index_z - 3, 0)
        index_to = min(index_z + 3, len(orbit_z_vec))
        z_around_crossing = orbit_z_vec[index_from : index_to]
        time_around_crossing = orbit_time_vec[index_from : index_to]

        # Set up spline interpolator and interpolate the time when z is equal to 0.0
        interpolator_time = InterpolatedUnivariateSpline(z_around_crossing,
                                                         time_around_crossing,
                                                         k=1)
        t_interp = interpolator_time(0.0)

        # Convert the interpolated time into datetime
        # Add the ascending node crossing time if it's before the sensing time
        datetime_ascending_crossing = (datetime_orbit_ref
                                       + datetime.timedelta(seconds=float(t_interp)))
        if datetime_ascending_crossing < sensing_time:
            datetime_ascending_node_crossing_list.append(datetime_ascending_crossing)

```

Reply:
I  understand this suggestion as to the improvement of the variable initialization and the readability improvement. I've taken another approach to achieve this.

Reply:
Early return has implemented as discussed in the RTC meeting :)